src/multi_dict.jl

```diff
@@ -127,3 +141,32 @@ function iterate(e::EnumerateAll, s)
     v, vst = vstate
     return ((k, v), (dstate, k, vs, vstate))
 end
+
+# grow_to! copied from Base -- needed for abstract generator constructor
+function Base.grow_to!(dest::MultiDict{K, V}, itr) where V where K
+    y = iterate(itr)
+    y === nothing && return dest
+    ((k,v), st) = y
+    dest2 = empty(dest, typeof(k), typeof(v))
+    insert!(dest2, k, v)
+    Base.grow_to!(dest2, itr, st)
+end
+
+# this is a special case due to (1) allowing both Pairs and Tuples as elements,
+# and (2) Pair being invariant. a bit annoying.
+function Base.grow_to!(dest::MultiDict{K,V}, itr, st) where V where K
+    y = iterate(itr, st)
+    while y !== nothing
+        (k,v), st = y
+        if isa(k,K) && isa(v,V)
+            insert!(dest, k, v)
+        else
+            new = empty(dest, promote_typejoin(K,typeof(k)), promote_typejoin(V,typeof(v)))
```

Comment:
It's trivial but use a better variable name than `new`. IMHO, it's better to avoid keywords as variable names. Probably, `new_md` would be much better.

Reply:
Eww gross, completely agreed. I copied this whole function from Base; sorry i didn't read it very closely.. Lemme make all your edits! 👍

Reply:
Done, thanks! :)